pkg/prometheus/operator.go

```diff
@@ -2081,3 +2121,14 @@ func validateRemoteWriteSpec(spec monitoringv1.RemoteWriteSpec) error {
 
 	return nil
 }
+
+func validateRelabelConfig(rc monitoringv1.RelabelConfig) error {
+	if _, exists := operator.ValidRelabelActions[rc.Action]; !exists {
+		return errors.Errorf("%s is an invalid relabel action", rc.Action)
+	}
+	// both replace and hashmod need targetLabel
+	if (rc.Action == "replace" || rc.Action == "hashmod") && rc.TargetLabel == "" {
```

Comment:
it would be nice to have constants for the know actions.

Reply:
yes, see https://github.com/prometheus/prometheus/blob/4c56a193c518ae6f56008b0a4c850a9c3f1477c6/model/relabel/relabel.go#L44-L56